filtered-resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/filteredresource/repository/RedisPersistenceContractTest.java

```diff
@@ -41,12 +48,15 @@
 import static java.util.Objects.requireNonNull;
 import static org.assertj.core.api.Assertions.assertThat;
 
-@SpringBootTest(
-        classes = {RedisPersistenceContractTest.class, FilteredResourceApplication.class},
-        properties = {"spring.data.redis.repositories.timeToLive.TokenOffsetEntity=1s"}
+@DataJpaTest(properties = {"spring.data.redis.repositories.timeToLive.TokenOffsetEntity=1s"})
+@ContextConfiguration(
+        classes = {ApplicationConfiguration.class, AsyncConfiguration.class, RedisTtlConfiguration.class, JpaTokenOffsetPersistenceLayer.class},
+        initializers = Initializer.class
 )
+@EnableAutoConfiguration
+@AutoConfigureDataRedis
+@AutoConfigureTestDatabase(replace = Replace.NONE)
 @ActiveProfiles("redis")
-@ContextConfiguration(initializers = Initializer.class)
 class RedisPersistenceContractTest {
```

Comment:
And this one as well
```suggestion
class RedisPersistenceComponentTest {
```